operators/pkg/tenant-controller/tenant_controller.go

```diff
@@ -320,22 +327,29 @@ func (r *TenantReconciler) createOrUpdateClusterResources(ctx context.Context, t
 	}
 
 	var retErr error
+	// retrieve instances
+	instances := crownlabsv1alpha2.InstanceList{}
+	err = r.Client.List(ctx, &instances, &client.ListOptions{Namespace: nsName})
+	if err != nil {
+		klog.Errorf("Unable to retrieve instances")
+		retErr = err
+	}
```

Comment:
Why are you retrieving the instances? All information should be present in the resource quota status.